driver-pravega/src/main/java/io/openmessaging/benchmark/driver/pravega/PravegaBenchmarkTransactionProducer.java

```diff
@@ -63,6 +64,9 @@ public PravegaBenchmarkTransactionProducer(String streamName, EventStreamClientF
     @Override
     public CompletableFuture<Void> sendAsync(Optional<String> key, byte[] payload) {
         try {
+            if (this.probeRequested(key)) {
+                this.probeTransaction(key, payload, this.eventsPerTransaction);
```

Comment:
Should this return the future here? `probeTransaction` returns `CompletableFuture.completedFuture(null)`, but seems to be never used. I think that this could be reworked as follows to prevent duplicating logic:

```
if (transaction == null) {
    transaction = transactionWriter.beginTxn();
}
if (this.probeRequested(key)) {
    // Populate transaction with the sufficient amount of events
    for (int i = 0; i < eventsPerTxn; i++) {
        transaction.writeEvent(key.get(), ByteBuffer.wrap(payload));
    }
}
```
Now, I assume that `++eventCount >= eventsPerTransaction` is going to be `true`, so the commit will be invoked, right? This means that we can get rid of `probeTransaction`  method. What do you think?